src/librustc_metadata/encoder.rs

```diff
@@ -1302,10 +1303,11 @@ impl<'a, 'b: 'a, 'tcx: 'b> IsolatedEncoder<'a, 'b, 'tcx> {
         }
     }
 
-    fn encode_info_for_ty_param(&mut self,
-                                (def_id, Untracked(has_default)): (DefId, Untracked<bool>))
-                                -> Entry<'tcx> {
-        debug!("IsolatedEncoder::encode_info_for_ty_param({:?})", def_id);
+    fn encode_info_for_generic_param(
+        &mut self,
+        (def_id, Untracked(encode_type)): (DefId, Untracked<bool>),
+    ) -> Entry<'tcx> {
+        debug!("IsolatedEncoder::encode_info_for_generic_param({:?})", def_id);
         let tcx = self.tcx;
         Entry {
             kind: EntryKind::Type,
```

Comment:
This seems wrong, it should either vary with the kind of parameter, or be something like `EntryKind::GenericParam`.

Reply:
It's only recording any concrete types (e.g. defaults or a const's type), so I think `EntryKind::Type` is correct. But the name is misleading; I'll rename it to `encode_info_for_ty_in_generic_param` or something like that.

Reply:
A `static`/`const` item also has a type, but doesn't use `EntryKind::Type`.
IMO we should treat generic parameters as proper definitions.